pkg/kubelet/kubelet.go

```diff
@@ -2248,22 +2249,40 @@ func (kl *Kubelet) rejectPod(pod *api.Pod, reason, message string) {
 // otherwise there will inevitably be pod delete create loops. This will be fixed
 // once we can extract these predicates into a common library. (#12744)
 func (kl *Kubelet) canAdmitPod(pods []*api.Pod, pod *api.Pod) (bool, string, string) {
-	if hasHostPortConflicts(pods) {
-		return false, "HostPortConflict", "cannot start the pod due to host port conflict."
-	}
-	if !kl.matchesNodeSelector(pod) {
-		return false, "NodeSelectorMismatching", "cannot be started due to node selector mismatch"
-	}
-	cpu, memory := kl.hasInsufficientfFreeResources(pods)
-	if cpu {
-		return false, "InsufficientFreeCPU", "cannot start the pod due to insufficient free CPU."
-	} else if memory {
-		return false, "InsufficientFreeMemory", "cannot be started due to insufficient free memory"
-	}
+	node, _ := kl.kubeClient.Core().Nodes().Get(kl.nodeName)
```

Comment:
The only thing I can think to do is to have Kubelet use the information it has locally to manufacture a Node object that looks like the one it would have received from the API server, and use that.